test/services/charts/daily_statistic_generator_test.rb

```diff
@@ -87,24 +89,24 @@ class DailyStatisticGeneratorTest < ActiveSupport::TestCase
 
     test "it should get block_timestamp" do
       block_timestamp_temp = Block.created_after(@started_at).created_before(@ended_at).recent.pick(:timestamp)
-      block_timestamp = Charts::DailyStatisticGenerator.new(@datetime).block_timestamp
+      block_timestamp = Charts::DailyStatisticGenerator.new(@datetime).call.block_timestamp
```

Comment:
Calling the call method directly will recalculate all attributes, which can be placed in the setup

Reply:
Charts::DailyStatisticGenerator.new(@datetime).call repeated calls too many times, unnecessary

Reply:
Because the test case have some special code before we call DailyStatisticGenerator.So we can't direct call in setup.